src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -256,6 +255,16 @@ private bool DynamicDataAttached(IDictionary<string, object?> sourceLevelParamet
             return false;
         }
 
+        if (test.TestMethod.DataType == DynamicDataType.None)
+        {
+            return false;
+        }
+
+        // REVIEW: We started setting this value to that actual DataType earlier so we can skip this expensive code
```

Comment:
Reading the comment only I don't understand the problem. I will try to debug to suggest what's missing.

Reply:
Improved the comment, the idea is to tell the AssemlbyEnumerator which test has data attributes, so it does not have to waste time constructing TestMethodInfo (an expensive type that is meant for test Execution), only to see that there are no data driven attributes.  But not all code is happy about this change, so we use it in this place to see if we should immediately leave this methos. And if we should not leave the method, we set the value to what it would have been originally, and let the code figure it out.